src/open_inwoner/accounts/backends.py

```diff
@@ -314,3 +314,186 @@ def update_user(self, user: AbstractUser, claims: JSONObject):
+            "email": generate_email_from_string(
+                claim_values["pseudo_id"], domain="localhost"
+            ),
+        }
+
+        # Try company first, as the required fields are unambiguous
+        has_company_name = bool(claim_values["company_name"])
+        has_legal_entity = bool(claim_values["legal_entity_id"])
+        if has_company_name or has_legal_entity:
+            if not (has_company_name and has_legal_entity):
+                raise SuspiciousOperation(
+                    "eIDAS company claims must include both company name and legal entity ID. "
+                    f"Received: company_name={'present' if has_company_name else 'missing'}, "
+                    f"legal_entity_id={'present' if has_legal_entity else 'missing'}."
+                )
+
+            return base_create_kwargs | {
+                "login_type": LoginTypeChoices.eidas_company,
+                "company_name": claim_values["company_name"],
+                "eidas_company_id": claim_values["legal_entity_id"],
+            }
+
+        # Assume this is a person
+        person_kwargs = base_create_kwargs | {
+            "login_type": LoginTypeChoices.eidas_person_pseudo_id,
+            "first_name": claim_values["first_name"] or "",
+            "last_name": claim_values["family_name"] or "",
+        }
+
+        # Do we have a BSN? If so, add it and update login type to reflect this
+        if claim_values["bsn"]:
+            person_kwargs = person_kwargs | {
+                "bsn": claim_values["bsn"],
+                "login_type": LoginTypeChoices.eidas_person_bsn,
+            }
+
+        return person_kwargs
+
+    def filter_users_by_claims(self, claims):
+        claim_values = self._extract_eidas_claim_values(claims)
+        if not claim_values["pseudo_id"]:
+            return User.objects.none()
+
+        return User.objects.filter(eidas_pseudo_id=claim_values["pseudo_id"])
+
+    def create_user(self, claims: JSONObject):
+        if not (
+            create_kwargs := self._construct_user_creation_kwargs_from_claims(claims)
+        ):
+            return None
+
+        logger.info(
+            "Creating new eIDAS user", pseudo_id=create_kwargs["eidas_pseudo_id"]
+        )
+        user = User.objects.create_user(**create_kwargs)
+
+        return user
+
+    def update_user(self, user: User, claims: JSONObject):
+        if not (
+            create_kwargs := self._construct_user_creation_kwargs_from_claims(claims)
+        ):
+            return None
+
+        for attr, val in create_kwargs.items():
+            setattr(user, attr, val)
+
+        try:
+            user.full_clean()
+        except ValidationError as exc:
+            raise SuspiciousOperation(
+                "Unable to update user due to a validation error"
+            ) from exc
+        else:
+            logger.info(
+                "Updating new eIDAS user", pseudo_id=create_kwargs["eidas_pseudo_id"]
```

Comment:
```suggestion
                "Updating eIDAS user", pseudo_id=create_kwargs["eidas_pseudo_id"]
```
or was the "new" intentional?